lib/std/math/isnormal.zig

```diff
@@ -1,45 +1,48 @@
 const std = @import("../std.zig");
 const math = std.math;
 const expect = std.testing.expect;
-const maxInt = std.math.maxInt;
 
-// Returns whether x has a normalized representation (i.e. integer part of mantissa is 1).
+/// Returns whether x is neither zero, subnormal, infinity, or NaN.
 pub fn isNormal(x: anytype) bool {
     const T = @TypeOf(x);
-    switch (T) {
-        f16 => {
-            const bits = @bitCast(u16, x);
-            return (bits + (1 << 10)) & (maxInt(u16) >> 1) >= (1 << 11);
-        },
-        f32 => {
-            const bits = @bitCast(u32, x);
-            return (bits + (1 << 23)) & (maxInt(u32) >> 1) >= (1 << 24);
-        },
-        f64 => {
-            const bits = @bitCast(u64, x);
-            return (bits + (1 << 52)) & (maxInt(u64) >> 1) >= (1 << 53);
-        },
-        f128 => {
-            const bits = @bitCast(u128, x);
-            return (bits + (1 << 112)) & (maxInt(u128) >> 1) >= (1 << 113);
-        },
-        else => {
-            @compileError("isNormal not implemented for " ++ @typeName(T));
-        },
+    const TBits = std.meta.Int(.unsigned, @bitSizeOf(T));
+    if (@typeInfo(T) != .Float) {
+        @compileError("isNormal not implemented for " ++ @typeName(T));
     }
+
+    const increment_exp = 1 << math.floatMantissaBits(T);
+    const remove_sign = ~@as(TBits, 0) >> 1;
+
+    // Hack: 1 is added to the exponent, and if it overflows to 0 or becomes 1,
```

Comment:
Should this really be considered a "hack"? This kind of bit manipulation seems to be quite common in the implementation of maths functions. Just wouldn't want future contributors to see this and think it's something that needs improving, if we're happy with this approach.

Reply:
I guess you're right! Wording is important, I more meant like a bit-hack haha. I'll adjust the wording.